crates/vm/levm/src/opcode_handlers/push.rs

```diff
@@ -44,15 +44,26 @@ impl VM {
     }
 }
 
-fn read_bytcode_slice(current_call_frame: &CallFrame, n_bytes: usize) -> Vec<u8> {
-    current_call_frame
+fn read_bytcode_slice(current_call_frame: &CallFrame, n_bytes: usize) -> Result<Vec<u8>, VMError> {
+    let pc_offest = current_call_frame
+        .pc()
+        // Add 1 to the PC because we don't want to include the
+        // Bytecode of the current instruction in the data we're about
+        // to read. We only want to read the data _NEXT_ to that
+        // bytecode
+        .checked_add(1)
+        .ok_or(VMError::Internal(
+            InternalError::ArithmeticOperationOverflow,
+        ))?;
+
+    Ok(current_call_frame
         .bytecode
-        .get(current_call_frame.pc()..)
+        .get(pc_offest..)
```

Comment:
This should be `pc_offset` instead of `pc_offest`.

Reply:
Yikes! Thanks Maxi.

Reply:
Fixed in latest commit. https://github.com/lambdaclass/ethrex/pull/1407/commits/4064b9ce8d301265d6cee100bbd8b6821be03784